### PR TITLE
feat: add retest on failure monitor property

### PR DESCRIPTION
**Pull request by shahzad31**
add retest on failure monitor property !!

### CODE REVIEW

**Comment by devcorpio** [comment]
Overall looks good, but I have a few questions/statements:

- Does this property work via `synthetics.config` file?
- Does this property work through `monitor.use`?
- I believe we should update cli.ts adding the description of this new CLI flag, too

**Comment by shahzad31** [comment]
> Overall looks good, but I have a few questions/statements:
> 
> * Does this property work via `synthetics.config` file?
Yes it does, unit test has been updated to reflect that
> * Does this property work through `monitor.use`?
yes it does, unit test has been updated to reflect that
> * I believe we should update cli.ts adding the description of this new CLI flag, too
no, it's not a cli option

**Review by devcorpio — APPROVED** [review]
LGTM

File: __tests__/push/monitor.test.ts [hunk 1]
```diff
@@ -245,6 +245,7 @@ heartbeat.monitors:
   tags:
     - ltag1
     - ltag2
+  retestOnFailure: true
```
[the remark this answers is not in the document]
Comment (vigneshshanmugam) [reply __tests__/push/monitor.test.ts:248]: What happens to max_attempts? How does a user handle how many attempts before giving up?

**Pull request merged**